questionpy_sdk/webserver/app.py

```diff
@@ -100,25 +115,174 @@ async def repeat_element(request: web.Request) -> web.Response:
+    assert webserver.attempt_scored
+    renderer = QuestionUIRenderer(xml=webserver.attempt_scored.ui.content,
+                                  placeholders=webserver.attempt_scored.ui.placeholders,
+                                  seed=webserver.attempt_seed)
+    context = {
+        'question_html': renderer.render_formulation(attempt=webserver.last_attempt_data,
+                                                     options=display_options),
+        'options': display_options.model_dump(exclude={'context', 'readonly'}),
+        'form_disabled': True
+    }
+
+    if display_options.general_feedback:
+        context['general_feedback'] = renderer.render_general_feedback(attempt=webserver.last_attempt_data,
+                                                                       options=display_options)
+    if display_options.feedback:
+        context['specific_feedback'] = renderer.render_specific_feedback(attempt=webserver.last_attempt_data,
+                                                                         options=display_options)
+    if display_options.right_answer:
+        context['right_answer'] = renderer.render_right_answer(attempt=webserver.last_attempt_data,
+                                                               options=display_options)
+    return context
+
+
+@routes.get('/attempt')
+async def get_attempt(request: web.Request) -> web.Response:
+    webserver: 'WebServer' = request.app['sdk_webserver_app']
+    stored_state = webserver.state_storage.get(webserver.package_location)
+    question_state = json.dumps(stored_state) if stored_state else None
+    display_options = QuestionDisplayOptions.model_validate_json(request.cookies.get('display_options', '{}'))
+    attempt_started_dict = json.loads(request.cookies.get('attempt_started', '{}'))
+    attempt_scored_dict = json.loads(request.cookies.get('attempt_scored', '{}'))
+    webserver.last_attempt_data = json.loads(request.cookies.get('last_attempt_data', '{}'))
+    webserver.attempt_seed = int(request.cookies.get('attempt_seed', -1))
+
+    if not question_state:
+        return web.HTTPNotFound(reason="No question state found.")
+
+    if webserver.attempt_seed < 0:
+        webserver.attempt_seed = random.randint(0, 10)
+
+    if attempt_started_dict and attempt_scored_dict:
+        webserver.attempt_started = AttemptStarted.model_validate(attempt_started_dict)
+        webserver.attempt_scored = AttemptScoredModel.model_validate(attempt_scored_dict)
+        context = await get_attempt_scored_context(webserver, display_options)
+    else:
+        context = await get_attempt_started_context(webserver, question_state, display_options)
+        assert webserver.attempt_started
+
+    response = aiohttp_jinja2.render_template('attempt.html.jinja2', request, context)
+    set_cookie(response, 'attempt_started', webserver.attempt_started.model_dump_json())
+    set_cookie(response, 'attempt_seed', str(webserver.attempt_seed))
+    return response
```

Comment:
Nach genauerer Überlegung finde ich den Datenfluss hier doch etwas komisch.
* Gibt es einen Grund, dass die Daten an der `WebServer`-Instanz gespeichert werden? Wenn sie per Cookie an den nächsten Request weitergegeben werden, ist das doch nicht nötig?
* Dass hier nur `start_attempt` und `score_attempt` abgedeckt werden, finde ich nicht optimal. Wir sollten auch `get_attempt` abdecken, sowohl für einen gestarteten, aber noch nicht bewerteten, Attempt, als auch für einen schon bewerteten.

  Ideal wäre auch eine Save-but-don't-submit Aktion, wie in Moodle.
  
  Ich würde vorschlagen, in Cookies nur attempt state und score (inkl. scoring state) abzuspeichern, und dann vorm Rendern `get_attempt` falls nötig aufzurufen, um an das `AttemptModel` zu kommen. So ist es auch im Plugin (naja, bisher minus scoring state)